doc/release/platforms/README.cray

```diff
@@ -420,52 +489,120 @@ To use this combination of runtime layers:
    of the total locale memory being wasted.
 
 
-Special parameters for the Cray-specific runtime layers
--------------------------------------------------------
+Due to the use of hugepages in the ugni comm layer, tasking layers
+cannot use guard pages for stack overflow detection.  Qthreads tasking
+uses uses guard pages for its stack overflow detection, so when ugni
+communications and qthreads tasking are combined, task call stack
+overflow detection is turned off.  This occurs automatically and nothing
+needs to be done to enable it.  (Muxed tasking uses another technique to
+detect stack overflow, as described below, so it is not affected by this
+limitation.)
+
+There are a few special parameters recognized by the ugni communication
+layer:
+
+    Communication Layer Concurrency
+    -------------------------------
+
+    The CHPL_RT_COMM_CONCURRENCY environment variable tells the ugni
+    communication layer how much program concurrency it should try to
+    support.  This basically controls how much of the communication
+    resources on the NIC will be used by the program.  The default
+    value is the number of hardware processor cores the program will
+    use for Chapel tasks (CHPL_RT_NUM_HARDWARE_THREADS in the next
+    section).  Usually this is enough, but for highly parallel codes
+    that do a lot of remote references, increasing it may help the
+    performance.  Useful values for CHPL_RT_COMM_CONCURRENCY are in
+    the range 1 to 30.  Values specified outside this range are
+    silently increased or reduced so as to fall within it.
+
+
+    Communication Layer Concurrency and Slurm
+    -----------------------------------------
+
+    When slurm is used for job placement on Cray systems, it limits the
+    total NIC memory registration in order to allow for job sharing on
+    the compute nodes.  In our experience this limit is approximately
+    240 GiB.  The product of CHPL_RT_MAX_HEAP_SIZE and the communication
+    layer concurrency discussed above must be less than this.  The ugni
+    communication layer adjusts its heap size and concurrency defaults
+    to reflect this limit when slurm is used for job placement.  The
+    default heap size is reduced to 16 GiB.  The concurrency is computed
+    such that the product of heap size and concurrency is below 240 GiB.
+    Thus under slurm, the ugni communication layer can support programs
+    with very large heaps or programs that need a lot of communication
+    concurrency, but not programs that need both simultaneously.  Such
+    programs need to be run using ALPS for job placement instead of
```

Comment:
I'm not sure I would include this last sentence. There's nothing the user can do about this.